pkg/cmd/cli/cmd/rollback.go

```diff
@@ -0,0 +1,101 @@
+template of the specified deployment will included in the rollback.
+
+Examples:
+  $ %[1]s %[2]s --from=deployment-2
+  ...
+
+  $ %[1]s %[2]s --from=deployment-3 --includeStrategy=true
+  ...
+
+  $ %[1]s %[2]s --from=deployment-1 --output=json
+  ...
+`
+
+func NewCmdRollback(parentName string, name string, f *Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Rolls back a deploymentConfig to match a previous deployment.",
+		Long:  fmt.Sprintf(rollbackLongDesc, parentName, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			from := kubecmd.GetFlagString(cmd, "from")
+			if len(from) == 0 {
+				usageError(cmd, "The --from argument is required.")
+			}
+
+			includeTemplate := kubecmd.GetFlagBool(cmd, "include-template")
+			includeTriggers := kubecmd.GetFlagBool(cmd, "include-triggers")
+			includeStrategy := kubecmd.GetFlagBool(cmd, "include-strategy")
+			includeReplMeta := kubecmd.GetFlagBool(cmd, "include-replication-meta")
+
+			output := kubecmd.GetFlagString(cmd, "output")
+
+			osClient, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace := getOriginNamespace(cmd)
+
+			rollback := &deployapi.DeploymentConfigRollback{
+				Spec: deployapi.DeploymentConfigRollbackSpec{
+					From: kapi.ObjectReference{
+						Name: from,
+					},
+					IncludeTemplate:        includeTemplate,
+					IncludeTriggers:        includeTriggers,
+					IncludeStrategy:        includeStrategy,
+					IncludeReplicationMeta: includeReplMeta,
+				},
+			}
+
+			newConfig, err := osClient.DeploymentConfigs(namespace).Rollback(rollback)
+			if err != nil {
+				glog.Fatalf("Error generating deploymentConfig for rollback: %v", err)
+			}
+
+			if output == "json" {
+				json, encodeErr := latestapi.Codec.Encode(newConfig)
+				if encodeErr != nil {
+					glog.Fatalf("Couldn't interpret rollback response: %v", err)
+				}
+
+				printer := client.JSONPrinter{}
+				if err := printer.Print(json, out); err != nil {
+					glog.Fatalf("unable to print deploymentConfig JSON: %v [%s]", err, json)
+				}
+
+				return
+			}
+
+			if _, err := osClient.DeploymentConfigs(namespace).Update(newConfig); err != nil {
+				glog.Fatalf("Error performing rollback: %v", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringP("from", "", "", "The name of a deployment (replicationController) resource")
+	cmd.Flags().BoolP("include-template", "", true, "Include the previous deployment's pod template in the rollback")
+	cmd.Flags().BoolP("include-triggers", "", false, "Include the previous deployment's triggers in the rollback")
+	cmd.Flags().BoolP("include-strategy", "", false, "Include the previous deployment's strategy in the rollback")
+	cmd.Flags().BoolP("include-replication-meta", "", false, "Include the previous deployment's replicationController metadata in the rollback")
+	cmd.Flags().StringP("output", "o", "json", "Output the rolled back deploymenConfig to STDOUT in the specified format rather than performing a rollback")
```

Comment:
there are some methods like `BoolVar` that you can use here that might make this cleaner.  You give it the var you want to store the flag in so you could declare the empty Spec outside of the cmd object (and reference it in the cmd) then have flags parse directly into it and remove the GetFlagBool lines